src/sinol_make/util.py

```diff
@@ -0,0 +1,111 @@
+import glob, importlib, os, sys, subprocess, requests, tarfile
+
+def get_commands():
+	"""
+	Function to get an array of all available commands.
+	"""
+	commands_path = glob.glob(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'commands/*'
+        )
+	)
+	commands = []
+	for path in commands_path:
+		temp = importlib.import_module('sinol_make.commands.' + os.path.basename(path), 'Command')
+		commands.append(temp.Command())	
+	
+	return commands
+
+
+def check_if_project():
+	"""
+	Function to check if current directory is a project
+	"""
+	
+	cwd = os.getcwd()
+	if os.path.exists(os.path.join(cwd, 'config.yml')):
+		return True
+	return False
+
+
+def check_oiejq(path = None):
+	"""
+	Function to check if oiejq is installed
+	"""
+	if sys.platform != 'linux':
+		return False
+	
+	def check(path):
+		p = subprocess.Popen([path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+		out, err = p.communicate()
+		if p.returncode == 0:
+			return True
+		else:
+			return False
+		
+	if path is not None:
+		return check(path)
+		
+	if not check(os.path.expanduser('~/.local/bin/oiejq')):
+		return False
+	else:
+		return True
+
+
+def install_oiejq():
+	"""
+	Function to install oiejq, if not installed.
+	Returns True if successful.
+	"""
+	
+	if sys.platform != 'linux':
+		return False
+	if check_oiejq():
+		return True
+
+	if not os.path.exists(os.path.expanduser('~/.local/bin')):
+		os.makedirs(os.path.expanduser('~/.local/bin'))
+
+	request = requests.get('https://oij.edu.pl/zawodnik/srodowisko/oiejq.tar.gz')
+	if request.status_code != 200:
+		raise Exception('Couldn\'t download oiejq (oij.edu.pl status code: ' + str(request.status_code) + ')')
```

Comment:
Let's also add the link from which we tried to download, so the user can check whether it's something wrong with the site or with the tool.